File: scm-csharp/scheme/primitives/PrimitiveWindersGet.cs
namespace scheme;

class PrimitiveWindersGet : Primitive {
    public override string Name() => "%winders-get";
    public override string Info()
    {
        return
            "Syntax: (%winders-get)\n" +
            "Library: (scm core)\n" +
            "Description: Returns the current winder chain list used by dynamic-wind to track before/after thunks.\n" +
            "Example:\n" +
            "  (%winders-get) => ()";
    }
    public override object Apply(SourcePos? pos, object[] args) {
        CheckArgs(pos, args, 0, 0);
        return VM.Current?.winders ?? Value.NIL;
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveWindersGet.java
package scheme.primitives;
import scheme.*;

public class PrimitiveWindersGet extends Primitive {
    @Override public String name() { return "%winders-get"; }
    @Override public String info() {
        return "Syntax: (%winders-get)\n" +
               "Library: (scm core)\n" +
               "Description: Returns the current winder chain list used by dynamic-wind to track before/after thunks.\n" +
               "Example:\n" +
               "  (%winders-get) => ()";
    }
    @Override public Object apply(SourcePos pos, Object[] args) {
        checkArgs(pos, args, 0, 0);
        VM vm = VM.current.get();
        return vm != null ? vm.winders : Value.NIL;
    }
}
